infra/retry.py

```diff
@@ -0,0 +1,108 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Retry decorator. Copied from ClusterFuzz source."""
+
+import functools
+import inspect
+import logging
+import sys
+import time
+
+# pylint: disable=too-many-arguments,broad-except
+
+
+def sleep(seconds):
+  """Invoke time.sleep. This is to avoid the flakiness of time.sleep. See:
+    crbug.com/770375"""
+  time.sleep(seconds)
+
+
+def get_delay(num_try, delay, backoff):
+  """Compute backoff delay."""
+  return delay * (backoff**(num_try - 1))
+
+
+def wrap(retries,
+         delay,
+         function,
```

Comment:
We can simplify this here by not requiring `function`. We needed this in CF to get the full qualified function name for metrics reporting, which is not needed here. We can use something like `__qualname__` on function to get the name (and possibly clean up CF as well in the future).

Reply:
Done.